klippy/stepper.py

```diff
@@ -294,23 +294,19 @@ def parse_step_distance(config, units_in_radians=None, note_valid=False):
 # Stepper controlled rails
 ######################################################################
 
-# A motor control "rail" with one (or more) steppers and one (or more)
+# A motor control carriage with one (or more) steppers and one (or more)
 # endstops.
-class PrinterRail:
+class GenericPrinterCarriage:
     def __init__(self, config, need_position_minmax=True,
-                 default_position_endstop=None, units_in_radians=False):
-        # Primary stepper and endstop
-        self.stepper_units_in_radians = units_in_radians
+                 default_position_endstop=None):
+        self.config = config
```

Comment:
The convention is to never store a persistent reference to the "config" object.  (Because, once the config phase ends, calls to the config methods can cause surprising results.)  Is it possible to rework this so that we don't store a long-held reference?

Reply:
I've indeed refactored the code to not keep the reference to config object beyond initialization.